File: estruturas/morris/morris_plus.py
from morris import Morris


class MorrisPlus:
    def __init__(self, t: int):
        self.t = t
        self.morris = [Morris() for _ in range(0, self.t)]

    def adiciona(self):
        for i in range(0, self.t):
            self.morris[i].adiciona()

    def conta(self):
        sum = 0
        for i in range(0, self.t):
            sum += self.morris[i].conta()

        return sum // self.t

File: estruturas/morris/c++/morris_plus.cpp
#include "morris_plus.hpp"
#include <cstdlib>
#include <cmath>

MorrisPlus::MorrisPlus(int T) : T(T) {
  morris.resize(T, Morris());
}

void MorrisPlus::adiciona() {
  for (int i = 0; i < T; ++i)
    morris[i].adiciona();
}

int MorrisPlus::conta() {
  double sum = 0;
  for (int i = 0; i < T; ++i)
    sum += morris[i].conta();
  
  return floor(sum / T);
}